api/extension-guides/ai/ai-extensibility-overview.md

```diff
@@ -0,0 +1,146 @@
+---
+ContentId: e375ec2a-43d3-4670-96e5-fd25a6aed272
+DateApproved: 06/12/2025
+MetaDescription: Overview of how to extend the AI features in your Visual Studio Code extension by using the Language Model, Tools, and Chat APIs.
+---
+# AI extensibility in VS Code
+
+This article provides an overview of AI extensibility options in Visual Studio Code, helping you choose the right approach for your extension.
+
+VS Code includes powerful AI features that enhance the coding experience:
+
+- **Code completion**: Offers inline code suggestions as you type
+- **Agent mode**: Enables AI to autonomously plan and execute development tasks with specialized tools
+- **Chat**: Lets developers use natural language to ask questions or make edits in codebase through chat interfaces
+- **Smart actions**: Use AI-enhanced actions for common development tasks, integrated throughout the editor
+
+You can extend and customize each of these built-in capabilities to create tailored AI experiences that meet the specific needs of your users.
+
+## Why extend AI in VS Code?
+
+Adding AI capabilities to your extension brings several benefits to your users:
+
+- **Enhanced user experience**: Provide intelligent assistance tailored to your extension's domain
+- **Domain specialization**: Create AI features specific to a programming language, framework, or domain
+- **Improved developer productivity**: Enhance common developer tasks, like debugging, code reviewing or testing, with AI capabilities
+- **Extend chat capabilities**: Add specialized tools or assistants to the chat interface for more powerful interactions
+
+## Extend the chat experience
+
+### Language model tool
+
+Language model tools enable you to extend agent mode in VS Code with domain-specific capabilities. In agent mode, these tools are automatically invoked based on the user's chat prompt to perform specialized tasks or retrieve information from a data source or service. Users can also reference these tools explicitly in their chat prompt by #-mentioning the tool.
+
+To implement a language model tool, use the [Language Model Tools API](/api/extension-guides/ai/tools) within your VS Code extension. A language model tool can access all VS Code extension APIs and provide deep integration with the editor.
+
+**Key benefits**:
+
+- Domain-specific capabilities as part of an autonomous coding workflow
+- Automatic tool invocation in agent mode, depending on user intent
+- Deep VS Code integration through extension APIs
```

Comment:
through extension APIs - let's make this clearer by saying "your tool implementation can use VS Code API since it runs in the Extension Host process"